### PR TITLE
UP-4929 : Build unit test cases for org.apereo.portal.portlets.account

### PR DESCRIPTION
**Pull request by nisithdash**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] tests are included
-   [ ] documentation is changed or added
-   [ ] [message properties][] have been updated with new phrases
-   [ ] view conforms with [WCAG 2.0 AA][]

##### Description of change

Adding new unit test cases for org.apereo.portal.portlets.account package

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apereo.portal.portlets.account;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.*;
```
Comment (bjagg) [inline uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java]: Generally, we do not use wildcard imports

File: uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+ * <p>
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apereo.portal.portlets.account;
+
+import static org.junit.Assert.*;
```
Comment (bjagg) [inline uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java]: Generally, we do not use wildcard imports

File: uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                                            "attr_name_1",
+                                            "attr_name_2",
+                                            "attr_name_3"
+                                        })));
+
+        CURRENT_ATTRIBUTE_VALUES =
+                Collections.unmodifiableList(
+                        new ArrayList<>(Arrays.asList(new String[] {"attr_value_0"})));
+    }
+
```
Comment (bjagg) [inline uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java]: Interesting approach. I like your use of unmodifiedSet/List wrappers on your static collection members.

File: uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        PersonForm personForm = userAccountHelper.getForm(TEST_USER_NAME);
+        assertEquals(personForm.getId(), 10000l);
+        assertEquals(personForm.getUsername(), TEST_USER_NAME);
+        // validate person attributes
+        Map<String, StringListAttribute> personAttributes = personForm.getAttributes();
+        assertNotNull(personAttributes);
+        StringListAttribute attributeValues = personAttributes.get(TEST_ATTRIBUTE_NAME);
+        List<String> values = attributeValues.getValue();
+        assertEquals(values.get(0), CURRENT_ATTRIBUTE_VALUES.get(0));
+    }
```
Comment (cbeach47) [inline uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java]: Can you add a test for when a null username is passed into getForm?

File: uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    /** @return */
+    private List<Preference> generateAccountEditPreferences() {
+        List<Preference> accountEditPreferences = new ArrayList<>();
+        Preference preference = new Preference();
+        preference.setName("pref_name_0");
+        preference.setDescription("pref_description_0");
+        accountEditPreferences.add(preference);
+        return accountEditPreferences;
+    }
+}
```
Comment (cbeach47) [inline uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java]: There looks to be several methods that could be added to this class such as canDeleteUser and canEditUser (at least up to the first conditional), createPassword, getNewAccountForm, etc.

File: uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        List<String> values = attributeValues.getValue();
+        assertEquals(values.get(0), CURRENT_ATTRIBUTE_VALUES.get(0));
+    }
+
+    @Test
+    public void testValidateLoginToken() {
+        boolean isTokenValid =
+                userAccountHelper.validateLoginToken(TEST_USER_NAME, TEST_USER_LOGIN_TOKEN);
+        assertTrue(isTokenValid);
+    }
```
Comment (cbeach47) [inline uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java]: Could you add tests for when nulls are passed in for the username and token?

**Comment by ChristianMurphy** [comment]
I checked with @bjagg and @nisithdash offline, there are no additional changes requested. :white_check_mark: 
Merging now, thanks @nisithdash! :bowing_man:
